modules/tinymce/src/core/main/ts/delete/InlineFormatDelete.ts

```diff
@@ -47,9 +58,63 @@ const deleteCaret = (editor: Editor, forward: boolean): Optional<() => void> =>
   });
 };
 
+const isBrInEmptyElement = (elm: Element): boolean => {
+  const parentElm = elm.parentElement;
+  return !Type.isNull(parentElm) && Empty.isEmpty(SugarElement.fromDom(parentElm)) && NodeType.isBr(elm);
+};
+
+const isEmptyCaret = (elm: Element): boolean =>
+  CaretFormat.isEmptyCaretFormatElement(SugarElement.fromDom(elm));
+
+const createCaretFormatAtStart = (editor: Editor, formatNodes: Node[]): void => {
+  const startElm = editor.selection.getStart();
+  // replace <br> in empty node or existing caret at start if applicable
+  // otherwise create new caret format at start
+  const pos = isBrInEmptyElement(startElm) || isEmptyCaret(startElm)
+    ? CaretFormat.replaceWithCaretFormat(startElm, formatNodes)
+    : CaretFormat.createCaretFormatAtStart(editor, formatNodes);
+  editor.selection.setRng(pos.toRange());
+};
+
+const updateCaretFormat = (editor: Editor, updateFormats: Node[]): void => {
+  const missingFormats = Arr.difference(updateFormats, getFormatNodesAtStart(editor));
+  if (missingFormats.length > 0) {
+    createCaretFormatAtStart(editor, missingFormats);
+  }
+};
+
+const isSelectionAtStartOfContainer = (editor: Editor): boolean =>
+  editor.selection.getRng().startOffset === 0;
```

Comment:
This condition seems rather general. The `startOffset` will mean different things depending on what the `startContainer` is. Given that this is the condition that we are using to determine whether or not we should override the delete, we'll need to confirm that anything that passes this condition can run `updateCaretFormat` without unexpected impact.

Reply:
Tightened condition to 
```ts
rangeStartsAtStartOfTextContainer(rng) && rangeStartContainerParentIsFormatElement(editor, rng) && rangeEndsAtOrAfterEndOfStartContainer(rng);
```
This was what I intended target the override for. Tests are added in InlineFormatDeleteOverrideTest.

Reply:
Also tightened override condition for `refreshCaret` in `requiresRefreshCaretOverride`:
```
rng.collapsed && (rangeStartsAtTextContainer(rng) || editor.dom.isEmpty(rng.startContainer)) && !hasAncestorInlineCaretAtStart(editor);
```

Reply:
Integrated tests in InlineFormatDeleteOverrideTest into InlineFormatDeleteTest and a new InlineFormatPostShortcutDeleteTest to increase specificity, make new tests consistent with existing ones, and not require exporting tests just for testing purposes.